ip_cidr_set.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"fmt"
+	"net"
+)
+
+type ipCIDRSet struct {
+	ip4NetMaps []ipNetMap
+	ip6NetMaps []ipNetMap
+}
+
+func newIPCIDRSet(nets []*net.IPNet) *ipCIDRSet {
+	w := &ipCIDRSet{
+		ip4NetMaps: make([]ipNetMap, 0),
+		ip6NetMaps: make([]ipNetMap, 0),
+	}
+
+	for _, net := range nets {
+		w.addIP(net.IP, net.Mask)
+	}
+
+	return w
+}
+
+func (w *ipCIDRSet) getNetMaps(ip net.IP) (netMaps *[]ipNetMap) {
+	// nolint:gocritic
+	if ip.To4() != nil {
+		netMaps = &w.ip4NetMaps
+	} else if ip.To16() != nil {
+		netMaps = &w.ip6NetMaps
+	} else {
+		panic(fmt.Sprintf("IP (%s) is neither 4-byte nor 16-byte?", ip.String()))
+	}
+
+	return netMaps
+}
+
+func (w *ipCIDRSet) has(ip net.IP) bool {
+	netMaps := w.getNetMaps(ip)
+	for _, netMap := range *netMaps {
+		if netMap.has(ip) {
```

Comment:
What's the difference between this and the `net` `Contains` method?

Reply:
O(1) on the list of IPs vs O(n)

Reply:
Sounds reasonable, could I ask that you put some comments through this code as a sort of documentation. Reading the code verbatim is quite hard to parse for me at least

Reply:
Have you had a chance to look at it since you first made this comment?  I've already added several comments and made minor simplifications to some of the logic.